oaknational/Oak-Web-Application
# [ENG-795] log first user visit through consent client

## Description

Music year: [1958](https://www.youtube.com/watch?v=epCN0f7FTIY)

* Updates @oaknational/oak-consent-client to 2.0.1
  * This makes a request to log the first visit by a user through the consent client which is why the new config `userLogUrl` was needed
* Some refactoring to remove a provider that was only needed to make testing easier

## How to test

1. Open the developer console on the network tab in an incognito session
2. Go to https://deploy-preview-2578--oak-web-application.netlify.thenational.academy
4. You should see a request to `/v1/user`
5. Reload the page
6. You should not see another request to `/v1/user`

## Checklist
